docs/blog/2021-09-07-mocking-a-production-api.md

```diff
@@ -0,0 +1,300 @@
+said on the tin and very well, however it didn't have an easy way to specify the
+custom query parameters we needed to faithfully reproduce the API like returning
+results before or after a given `created_at` timestamp (feel free to let us know
+if we missed something here!).
+
+So we needed something a little more sophisticated. The internet came to the
+rescue again with a comprehensive
+[list](https://en.wikipedia.org/wiki/Comparison_of_API_simulation_tools) of API
+simulation tools. The basic precondition we had was that the API simulator had
+to be OSS with a permissive license we could build on. This immediately
+disqualified 50% of the available solutions, and we did a divide and conquer
+exercise quickly evaluating the rest.
+
+![Api Simulation Tools](media/api-simulation-tools.png)
+
+The remaining tools were either not built for this purpose, or they were
+incredibly complex pieces of software that would take a while to get acquainted
+with.
+
+In the end we decided to implement the endpoint functionality ourselves - we
+figured that a 50 LOC NodeJS server would do a fine job for a POC.
+
+## Day 4: Implementing the core functionality
+
+Day 4 was the most straight forward. Let's get this thing to work!
+
+### 1. The Data Model
+
+We decided to reproduce the API at level 2 since we didn't really have any other
+endpoints. We used [synth](https://github.com/getsynth/synth) to quickly whip up
+a [data model](TODO_link_to_GH) that generates data that looks like the Shopify
+API. I won't go into depth on how this works here as it's been covered
+in [other](2021-08-31-seeding-databases-tutorial.md) posts. In about 15 minutes
+we had 10 Mb data that looks like this:
+
+```json
+...
+{
+    "arguments": "generate virtual platforms",
+    "body": null,
+    "created_at": "2019-09-17T14:16:47",
+    "description": "Received new order",
+    "id": 477306,
+    "message": "Received new order",
+    "path": "/admin/orders/83672/transactions/#6020",
+    "subject_id": 1352997,
+    "subject_type": "Order",
+    "verb": "closed"
+},
+{
+"arguments": "innovate best-of-breed schemas",
+"body": null,
+"created_at": "2017-05-20T00:04:41",
+"description": "Received new order",
+"id": 370051,
+"message": "Received new order",
+"path": "/admin/orders/82607/transactions/#9154",
+"subject_id": 1226112,
+"subject_type": "Order",
+"verb": "sale_pending"
+},
+{
+"arguments": "incentivize scalable mindshare",
+"body": null,
+"created_at": "2018-02-21T12:51:36",
+"description": "Received new order",
+"id": 599084,
+"message": "Received new order",
+"path": "/admin/orders/01984/transactions/#3595",
+"subject_id": 1050540,
+"subject_type": "Order",
+"verb": "placed"
+}
+...
+```
+
+We then dumped it all in a MongoDB collection with the one-liner:
+
+```bash
+$ synth generate shopify --to mongodb://localhost:27017/shopify --size 40000
+```
+
+Next step is to re-create the Events API endpoint.
+
+### 2. Creating the API
+
+@brokad
```

Comment:
Maybe you can fill in the gaps for this: we just wrote a prisma model for the responses and it worked OOB with the data with dumped into mongo. Then we just wrote a quick and dirty express server that maps the REST endpoint's querystring into a query for prisma.